### PR TITLE
enhance: Append drop partition msg to wal

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/33285

### CODE REVIEW

File: internal/distributed/streaming/streaming.go [hunk 1]
```diff
@@ -9,7 +9,11 @@ import (
 	"github.com/milvus-io/milvus/pkg/streaming/util/options"
 )
 
-var singleton *walAccesserImpl = nil
+var singleton WALAccesser = nil
+
+func SetWAL(w WALAccesser) {
```
Comment (chyezh) [inline internal/distributed/streaming/streaming.go:14]: move it into a test-build only file?

File: internal/distributed/streaming/streaming.go [hunk 2]
```diff
@@ -19,8 +23,8 @@ func Init(c *clientv3.Client) {
 
 // Release releases the resources of the wal accesser.
 func Release() {
-	if singleton != nil {
-		singleton.Close()
+	if w, ok := singleton.(*walAccesserImpl); ok && w != nil {
```
Comment (jaime0815) [inline internal/distributed/streaming/streaming.go:26]: Suggest adding a close method to the WALAccesser interface and invoking it directly.

**Comment by jaime0815** [comment]
add create partition msg to wal in the next PR

**Pull request merged**